### ISSUE TITLE
When the duplicate-merging script deletes a sentence in the process of merging sentences, automatically rate the sentence OK by the person who contributed the duplicate.

### ISSUE THREAD

**Issue by ckjpn** [Tatoeba/tatoeba2#1016]
This may be too difficult to program, but the following might be something to consider.

When the duplicate-merging script deletes a sentence in the process of merging sentences, automatically rate the sentence OK by the person who contributed the duplicate.

By contributing the duplicate sentence, the second contributor is saying that he thinks the sentence is good, isn't he?

Here's an example.

Guybrush88, a native Italian speaker, contributes a sentences that is a duplicate of a non-native speaker's contribution.

From a user's point of view, it would be nice to know that Guybrush88 thought that that Italian sentence was good, so rate the sentence OK by Guybrush88.

I doubt that anyone would complain about this.  Why would someone intentionally contribute a sentence that they didn't feel was good?

**Comment by RyckRichards** [comment]
Actually, as far it's a collection of sentences that an user has,
maybe each user should "check" if they want this or not. What we can
do is to tag the sentence as OK since a native already posted. that
sentence.

On 2/23/16, ckjpn <notifications@github.com> wrote:
> This may be too difficult to program, but the following might be something
> to consider.
>
> When the duplicate-merging script deletes a sentence in the process of
> merging sentences, automatically rate the sentence OK by the person who
> contributed the duplicate.
>
> By contributing the duplicate sentence, the second contributor is saying
> that he thinks the sentence is good, isn't he?
>
> Here's an example.
>
> Guybrush88, a native Italian speaker, contributes a sentences that is a
> duplicate of a non-native speaker's contribution.
>
> From a user's point of view, it would be nice to know that Guybrush88
> thought that that Italian sentence was good, so rate the sentence OK by
> Guybrush88.
>
> I doubt that anyone would complain about this.  Why would someone
> intentionally contribute a sentence that they didn't feel was good?
>
>
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/Tatoeba/tatoeba2/issues/1016


-- 

Ricardo Vernaut Jr

Mobile // Celular: (5521) 967-640-668
Email: rvernautjr@gmail.com
<https://www.linkedin.com/in/vernautjr>

**Comment by trang** [comment]
I've changed the title.

The idea itself is not a wrong idea, although it would need some discussion, but it's not the job of the duplicate-merging script and won't be implemented in the duplicate-merging script.

@ckjpn, I know it's probably a habit that is difficult to get rid of, but when creating issues, remember to focus on the problem first, and suggest a solution only after (cf. the [guidelines][1]). You also sometimes create very long titles when they could be much shorter.

[1]: https://github.com/Tatoeba/tatoeba2/wiki/Guidelines-for-submitting-issues

**Comment by ckjpn** [comment]
OK.  The problem is that a native speaker takes to the time to contribute a good sentence, but it doesn't really help us know that it's a good sentence because instead of having a sentence written by a native speaker, we still have a non-native speaker contribution that isn't rated OK by a native speaker and the native-speaker sentence gets thrown away.

**Comment by sacredceltic** [comment]
I agree with CK that it's a real issue. Still, we have to agree to what a "native" speaker exactly is...

**Comment by ckjpn** [comment]
Even if it's not a "native speaker" but we trust the person's opinion on correctness for that language, it's useful information to have.

From my point of view, as someone who uses the Tatoeba Corpus for other projects, it's useful for me to know this information since I can feel comfortable using an Italian sentence contributed by Guybrush88 or rated OK by Guybrush88, for example.  It's up to whoever uses the data to make a decision on who they are willing to trust, I think.

**Comment by ckjpn** [comment]
Note that this is likely an even better idea now that ratings (now called collections) can be added by more than one member.

Perhaps if this is not difficult to do, it would be a good idea to add it now.

**Comment by trang** [comment]
@alanfgh expressed some concerns regarding adding such ratings automatically (see [his comment on a recent pull request that attempted to implement this](https://github.com/Tatoeba/tatoeba2/pull/2598#issuecomment-749138151)).

I'll reply to this comment here.

> In general, there is only one situation in which I believe that "OK" ratings, tags, or lists are beneficial, and that is when it is important to indicate that the quality of a sentence was previously in question, but no longer is (either because the sentence has been fixed or because someone has said that the sentence is fine as it is). 

I agree with the principle, but how do you define a sentence that was previously in question? It seems to depend on the level of trust (or mistrust) of the user. For a large number of users, the quality of a sentence added by a non-native speaker is in question the moment it is added in Tatoeba. And for some users, it may not matter whether the sentence was from a native speaker or not, the doubt on the quality will depend if the sentence was added by someone who is in their circle of trust.

> Yet, since simple sentences are more likely to be duplicated, they would end up accumulating more "OK" ratings when the duplicates were merged.

Indeed, this could create a lot of clutter. Plus, these sentences may already have other indicators that they are OK (for instance, they have audio, or have hundreds of translations). I can imagine that a sentence like "I love you." could get added as a duplicate dozens of times, and therefore would end up with dozens of "OK" reviews that have little to no value.

It would be interesting if we implemented something to track sentences that were never effectively added because they were a duplicate of an existing sentence. First, we could see how often that happens. Then we can see concrete examples where having an automatic OK review would have been beneficial or on the contrary, would have just been extra clutter.

> I'm not sure I understand the "salvaging wasted time" argument. The best way not to waste one's time is to write sentences that are unlikely to be duplicates in the first place. As for correcting sentences that thereby become duplicates, the "win" in terms of consolidating them is self-evident.

It's more about trying to salvage information than trying to salvage wasted time. But I can see how the information might bring more clutter than usefulness.

**Comment by alanfgh** [comment]
@trang Thanks for putting my comment and your feedback (which I agree with) here and drawing my attention to this issue ticket. Now I understand better the context for the proposal, and what it's trying to accomplish. I see that it's meant to address the situation where a contributor has already added a sentence that a "more trusted" contributor (for instance, a native speaker where the original contributor was non-native) now wants to add (possibly as a new sentence, possibly by correcting an existing sentence). But there is no logic here that would take into account whether the new sentence contributor is more trusted than the old one, so we'd end up adding an "OK" rating in any case. This could, as Trang says, add to clutter. It could also, as I've pointed out, exacerbate the problem where "OK" ratings are handed out preferentially to the kinds of simple sentences that already get more attention (audio, translations, and so on), leading to a runaway effect that decreases the diversity and usefulness of the corpus.
